src/zbeacon.c

```diff
@@ -654,3 +711,135 @@ zbeacon_test (bool verbose)
     //  @end
     printf ("OK\n");
 }
+
+
+//  --------------------------------------------------------------------------
+//  Selftest
+
+void zbeacon_ipv4_mcast_test (bool verbose)
```

Comment:
Don't add a new function, fold it in the existing one. Also, you need to do a check if setting v4 multicast addresses is allowed on the platform at all - it might not, and that's not a reason to fail the test. For example, all CI runs are broken due to this right now. This needs to be fixed before it can be merged.

Reply:
Is it OK if I just chain this off the existing function? i.e. call this at the end of the existing test?

Reply:
yes

Reply:
For your other request to check if setting the address is allowed, is there a special function for this? Looking around on the internet everyone is just suggesting running ifconfig or netstat -g.

I am not seeing anything programatically showing how to check this.

Reply:
What we usually do in unit tests is simply trying to set it, and fail gracefully (hiding the error) if it doesn't work

Reply:
Man you are fast! Which is quite nice when trying to wrap up stuff like this 😄 

As for trying to fail gracefully I am struggling with that. The fact that the beacon is run on a pipe/thread with only one way signaling makes it very hard to trickle any status stuff back out also all the function calls like s_self_prepare_udp return void so its difficult to even call those directly and check for failure that way.

I can try to make a socket directly using zsys_udp_new (enable_multicast) and set the flags on that socket, but that isn't really testing anything it would just cause it to abort running the "real" code. Also this would duplicate a huge chunk of the logic in s_self_prepare_udp as most of that logic is required to get to a state where you are ready to call set multicast.

Do you have any suggestions on a better/ideal approach?

Reply:
Copy a shortened implementation of the bsd socket creation with all the multicast flags, and run it before the test. If it works, then the test can go ahead, otherwise it can be skipped. This is how we test for ipv6 support for example.